### ISSUE TITLE
inconsistent and unclear information on new browser for webmap preview 

### ISSUE THREAD

**Issue by Jeroen-GroeneBij** [qgis2web/qgis2web#1110]
Hi,

First of all a big thank you for picking up the development of this plugin.

My question is about the message for the preview functionality. The message reads: 
> qgis2web supports a new browser for webmap preview, but it is not available because the PyQtWebEngine 5.15.6 dependency is missing or your QGIS version is earlier than 3.38.0
> 
> Please ensure you have:
> - QGIS >= 3.38.0
> - AN ACTIVE INTERNET CONNECTION (even under proxy)
> Then click the button to install.
> 
> WARNING: QGIS will close after installation, SAVE YOUR PROJECT FIRST.
> 
> PS: In absence of this webmap preview, you can still use qgis2web to Export your project.

Now, the thing is, it refers to two different pieces of software: QGIS version 3.38.0 and up and  PyQtWebEngine 5.15.6 and up.
And then it gives me a choice to keep using the old browser or the new one.
I read the message as: hey, I would expect to find the dependency for the new browser, but your installation does not have this yet. Click here to install the new browser.

However, I am using QGIS 3.38.3, on a Windows 10 machine.
So why does this installation not have that particular dependency?

The reference to PyQtWebEngine 5.15.6 does not make this more clear for me.
In the QGIS About page, there is no mentioning of PyQtWebEngine. There is mentioning of Qt version (5.15.13) and a Python version (3.12.6)
Is the PyQtWebEngine in any form related to the Qt version?
If so, should the new browser by default be available and usable when using QGIS 3.38.3? So, without the need use the install button?

If there is indeed a direct relation between PyQtWebEngine and Qt version, than please note that 3.34.11, the latest LTR also uses QT 3.15.13 So is there indeed a need for 3.38.0 and up?

I also noticed in another issue (1092) that QGIS 3.38.2 uses Qt 3.15.2 which could explain the missing dependency. If, so the minimum QGIS version of 3.38.0 is incorrect.

So, in a nutshell. I am confused because I get this message, even when using 3.38.3
And to add to my own confusion I am making the assumption that the PyQtWebEngine and the Qt version have a direct relationship.

Oh and while writing this I just thought of something:
Might the missing dependency have anything to do with me having multiple QGIS versions installed and using multiple QGIS version and thus using 'existing' profiles?

**Comment by andreaordonselli** [comment]
Thanks for your message.
When you install QGIS 3.38.3, the PyQtWebEngine (that we can define it as a browser engine) dependency is not installed; it must be installed by hand and will only work if there are other components that are only installed from QGIS 3.38 onwards.
It doesn't matter that you have previous versions of QGIS installed in your system.

I put the code in a "try" block so if a new Qgis version will come out that integrates PyQtWebEngine The message will no longer be shown at startup, qgis2web will automatically start with the new technology.

That said, If you have any advice on how to improve the message please write them, we will consider them.
The message written today is created so as not to install anything without the user's knowledge but to allow him to choose, and at the same time explain well what is missing and what will be installed.

I hope qgis2web meets your expectations and thank you for raising the question.
Regards
Andrea

**Comment by workflowsguy** [comment]
Hello,

I have the same issue as OP.

macOS 11.7.10
QGIS 3.40.1
qgis2web 3.23.0

I have installed qt5-qtwebengine @5.15.17_1 through MacPorts, but the Plugin's preview window still shows the message
```
qgis2web supports a new browser for webmap preview, but it is not available because the PyQtWebEngine 5.15.6 dependency is missing or your QGIS version is earlier than 3.38.0

Please ensure you have:
- QGIS >= 3.38.0
- AN ACTIVE INTERNET CONNECTION (even under proxy)
Then click the button to install.

WARNING: QGIS will close after installation, SAVE YOUR PROJECT FIRST.

PS: In absence of this webmap preview, you can still use qgis2web to Export your project.
```

instead of the actual preview.

Also, when I then click on the button "Install new browser", an error dialog 

"An unexpected error occurred: [Errno 2] No such file or directory: 'brew'" 

appears even though the `brew` command is installed.

**Comment by andreaordonselli** [comment]
Try installing pyqtwebengine 5.15.6
Unfortunately I can't test on Mac, so I can't tell you why brew doesn't work
